### ISSUE TITLE
New Zealand CO2 Emissions Data Source has been updated...

### ISSUE THREAD

**Issue by ChirpyTurnip** [electricitymap/electricitymap-contrib#3562]
Hi,

The NZ source of the CO2 data has been updated. The news release is here: https://www.ems.co.nz/news/carbon-tracker/

The new data source is here: https://app.em6.co.nz/

Is it possible to update the scraper so that the data once again appears on the map? 

Cheers!

CP.

**Comment by drc38** [comment]
It looks like the parser uses transpower's live data as the main source https://www.transpower.co.nz/power-system-live-data
and that it no longer provides North and South Island data, but rather "New Zealand" combined.
```
"},"soPgenGraph":{"data":{"New Zealand":{"Battery":{"generation":-0,"capacity":1},"Co-Gen":{"generation":138,"capacity":216},"Coal":{"generation":0,"capacity":750},"Gas":{"generation":385,"capacity":1280},"Geothermal":{"generation":876,"capacity":1062},"Hydro":{"generation":3735,"capacity":5415},"Liquid":{"generation":0,"capacity":156},"Wind":{"generation":116,"capacity":1040},"total":5249.406}},"date":"24 Nov 2021 10:00","timestamp":1637701200,"update":true},"soRenewablePgenGauge":{"data":[90.04],"date":"24 Nov 2021 10:00","timestamp":1637701200,"update":true}
```
I will have a go modifying the parser based on the above.

**Comment by madsnedergaard** [comment]
Would it be possible to use some of the other API endpoints in order to maintain the difference between South and North? 

Merging the two zones will be hard due to our infrastructure and existing data, so would be great if we could find another way to do it :)

https://www.ems.co.nz/em6-api-integration-guide/

**Comment by drc38** [comment]
The em6 service is a paid for api, would the project be able to cover the cost?

**Comment by ChirpyTurnip** [comment]
Do we really need the difference between north island and south island? I'm
not even sure if the integration asked me what island I was on when it was
added to home assistant. And, given that the HVDC link could be be flowing
in either direction, or in both direction simultaneously (yes, it does,
it's called 'round power') the difference is mostly academic. And having an
indication of overall renewable content is better than none.

Alternatively we could ask Transpower for a feed specifically for what we
need. I have a contact there....there's no harm in asking. If so, someone
will need to provide me with a technical blurb to pass on to them...

CP.

**Comment by jarek** [comment]
> Do we really need the difference between north island and south island?

Well, we previously had this data split between the islands, and it did indeed show a difference: South Island is almost entirely hydroelectric power, while North usually had a mix of hydro, geothermal, and some fossil/gas if I recall correctly. As I recall, hydro in the South was being sometimes used as a storage facility by reducing hydro production and importing from North when North had plenty.

> And, given that the HVDC link could be be flowing in either direction, or in both direction simultaneously (yes, it does, it's called 'round power') the difference is mostly academic.

The difference that was seen was that South Island's electricty was considered less CO2eq-intensive than the North's, but became relatively more CO2eq-intensive when it was importing from the North.

But all other things being equal, more detailed data is better than less detailed data.

> Alternatively we could ask Transpower for a feed specifically for what we need. I have a contact there....there's no harm in asking. If so, someone will need to provide me with a technical blurb to pass on to them...

Basically we would want a per-fuel/source breakdown as available for whole country on https://www.transpower.co.nz/power-system-live-data (Battery, Co-Gen, Coal, Gas, Geothermal, Hydro, Liquid, Wind), except separated out into generation on North and South Island for each of these.

While we're at it - does anyone know what the "Liquid" source is? Is it fuel oil/diesel?

**Comment by drc38** [comment]
>While we're at it - does anyone know what the "Liquid" source is? Is it fuel oil/diesel?

@jarek most likely diesel gensets

**Comment by ChirpyTurnip** [comment]
Hi,

After some investigation it seems that Transpower will no longer be
offering island or region based generation mix data. They believe that this
provides a misleading picture as to how 'green' an island or region is,
when the reality is that as a power system it would not be possible to
operate without the other fuel types in the system - therefore we must
consider the 'complete' picture.

There is a nice guy who works there who has offered to talk a developer if
you wish to discuss the possibility of being able to re-point to another
API for the purposes of getting this data. On the basis that the service
provided by electricitymap / CO2Signal is not-for-profit it sounded to me
like there is a possibility of getting free access to at least some of the
em6 API...which I think would be really useful, and more stable than the
System operator website as the EM6 interface is used commercially.

His contact details are:

Edward Oosterbaan
Senior Business Analyst
EMS Operations - Unregulated
Transpower NZ Ltd

E: edward.oosterbaan |at| transpower.co.nz

CP.

**Comment by ChirpyTurnip** [comment]
Also....the existing em6 service is also about to go away as that is all
deprecated. I'm pretty sure the new service is now / going here:
https://app.em6.co.nz/

**Comment by skovhus** [comment]
We were contacted by EMS with the following message:

“
We have moved away from reporting emissions and generation by Island because this was misleading our end users.

You can source real time carbon emissions data and renewal percentages here:

https://api.em6.co.nz/ords/em6/data_api/current_carbon_intensity
“

**Comment by drc38** [comment]
Can the PR be reviewed and merged on that basis?

**Comment by madsnedergaard** [comment]
I have reached out to the contact shared by @ChirpyTurnip. We unfortunately can't just merge the PR @drc38, as it will require some additional work as I mentioned earlier - also, we'd rather not lose the granularity and detail if possible :)

**Comment by drc38** [comment]
@madsnedergaard ok keep us posted, it'd be good to have the feed back up and running.

**Comment by luke-barnett** [comment]
> I have reached out to the contact shared by @ChirpyTurnip. We unfortunately can't just merge the PR @drc38, as it will require some additional work as I mentioned earlier - also, we'd rather not lose the granularity and detail if possible :)

Is this not incorrect data unless the HVDC link is also taken into account (which as mentioned is complicated)? It might make sense from a generation point of view, but from a consumption (which is at least what I'm using this data for) it's far more complicated and potentially misleading.

@ChirpyTurnip @drc38 I've snooped a little into the free EM6 dashboard: and it looks like we could get something (with a bit of manual manipulation) using this endpoint:

https://api.em6.co.nz/ords/em6/data_api/free/price

The grid_zone_id parameter can then be used to filter that down and then group them back up to North/South.

https://api.em6.co.nz/ords/em6/data_api/free/price?grid_zone_id=1

It would be pretty error-prone if we don't also have a feed for the zone ids. The other key difference is that the data is historical :(

**Comment by drc38** [comment]
> @ChirpyTurnip @drc38 I've snooped a little into the free EM6 dashboard: and it looks like we could get something (with a bit of manual manipulation) using this endpoint:

> https://api.em6.co.nz/ords/em6/data_api/free/price

Interesting, what other end points have you found? This one is not listed in the api integration document.

**Comment by luke-barnett** [comment]
> > @ChirpyTurnip @drc38 I've snooped a little into the free EM6 dashboard: and it looks like we could get something (with a bit of manual manipulation) using this endpoint:
> 
> > https://api.em6.co.nz/ords/em6/data_api/free/price
> 
> Interesting, what other end points have you found? This one is not listed in the api integration document.

https://api.em6.co.nz/ords/em6/data_api/price/24hrs/OTA2201,WKM2201,HAY2201,KIK2201,ISL2201,BEN2201

This seems to give you the historical pricing (MW not MWh) for the last 24 hours of trading periods, 48 half hours.

https://api.em6.co.nz/ords/em6/data_api/free/price

Gives you the historical generation values per generation source.

https://api.em6.co.nz/ords/em6/data_api/region/price

Gives you the regional pricing for the previous trading period.

So it all seems historical and not the current state so I don't believe they are exactly what we are after but it could be better than nothing?

**Comment by madsnedergaard** [comment]
> @madsnedergaard ok keep us posted, it'd be good to have the feed back up and running.

Hey again, I've been in contact with Edward from Transpower, and he shared the following about why they would prefer to not split the zones:
> We would prefer to show the generation for the whole country as the islands are in the same network. There is something called reserve sharing between the islands where generation is only possible if it is backed by standby generation in another island. This means the grid can really only be considered to be a single grid. The new generation data is much more accurate and takes into account private data that only we have access to. If we were to split it into islands the private hidden data would be visible. (previously this information was not included in what you were scraping)

So in conclusion, I think the best solution going forward is making the necessary changes to our API and co2signal API - and then merge https://github.com/electricityMap/electricitymap-contrib/pull/3567 :)

I'll post any new updates on this issue.

**Comment by luke-barnett** [comment]
@madsnedergaard any better sources for cost? Those are currently being scraped from the old EM6 dashboard which I'm guessing is due to disappear with the new one.

It is wholesale and not retail pricing and NZ does have regional pricing depending on roughly where you live, so it does make sense to separate that by island as it can vary dramatically.

It probably doesn't make much sense to have one dataset split for one and combined for the other?

**Comment by drc38** [comment]
> @madsnedergaard any better sources for cost? Those are currently being scraped from the old EM6 dashboard which I'm guessing is due to disappear with the new one.
> 
> It is wholesale and not retail pricing and NZ does have regional pricing depending on roughly where you live, so it does make sense to separate that by island as it can vary dramatically.
> 
> It probably doesn't make much sense to have one dataset split for one and combined for the other?

I've updated the Ems source to the new api in PR #3567

**Comment by luke-barnett** [comment]
> > @madsnedergaard any better sources for cost? Those are currently being scraped from the old EM6 dashboard which I'm guessing is due to disappear with the new one.
> > It is wholesale and not retail pricing and NZ does have regional pricing depending on roughly where you live, so it does make sense to separate that by island as it can vary dramatically.
> > It probably doesn't make much sense to have one dataset split for one and combined for the other?
> 
> I've updated the Ems source to the new api in PR #3567

Based on what I could read from the dataset, it's technically historical pricing? As it is labelled as being a trading period behind. It probably doesn't matter in the scheme of things as the em6live dashboard could have been the same and been none the wiser.

**Comment by drc38** [comment]
@madsnedergaard is a release scheduled soon to get NZ back online?

**Comment by madsnedergaard** [comment]
Hey @drc38, sorry for the lack of update - we hope to get it out today if we can get our CI running again :)

**Comment by ChirpyTurnip** [comment]
I see data!!!!  Yay!

**Comment by madsnedergaard** [comment]
We finally managed to get things fixed, so NZ is now on the map! Thanks for the work here 💪

**Comment by ChirpyTurnip** [comment]
Unfortunately it looks like it is broken again. Died in the last couple of days....on 16/3 I was showing 79.62% renewable, on 17/3 just 27.12%, and from 18/3 onwards back to 2% - 10%.....

So I suspect EMS have changed the web site /API again because the data isn't coming through anymore. :-(

**Comment by drc38** [comment]
> Unfortunately it looks like it is broken again. Died in the last couple of days....on 16/3 I was showing 79.62% renewable, on 17/3 just 27.12%, and from 18/3 onwards back to 2% - 10%.....
> 
> So I suspect EMS have changed the web site /API again because the data isn't coming through anymore. :-(

See #3870, should be fixed with next release
